plugin-gradle/src/main/java/com/diffplug/gradle/spotless/KotlinExtension.java

```diff
@@ -58,26 +60,31 @@ public LicenseHeaderConfig licenseHeaderFile(Object licenseHeaderFile) {
 	}
 
 	/** Adds the specified version of <a href="https://github.com/pinterest/ktlint">ktlint</a>. */
-	public KotlinFormatExtension ktlint(String version) {
+	public KotlinFormatExtension ktlint(String version) throws IOException {
 		Objects.requireNonNull(version);
-		return new KotlinFormatExtension(version, false, Collections.emptyMap(), Collections.emptyMap());
+		File defaultEditorConfig = getProject().getRootProject().file(".editorconfig");
+		FileSignature editorConfigPath = FileSignature.signAsList(defaultEditorConfig);
```

Comment:
This leads to I/O during the Gradle configuration phase, and IIRC it's generally not recommended. Is there any way to do this in a deferred fashion?

Reply:
The same is used for `Diktat` and specifically requested by @nedtwigg in the comment above

Reply:
Fair point. Maybe something for the future. No premature optimisation.